### PR TITLE
[DO NOT MERGE] treewide: Initial role commit

### PR DESCRIPTION
**Pull request by haslersn**
- [ ] `README.md`
- [x] Tested to be working
- [x] Fix overmodulation
- [ ] Document https://github.com/haslersn/btrust

### CODE REVIEW

File: files/btrust.service [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Btrust
+After=bluetooth.service
+Wants=bluetooth.service
+
+[Service]
+ExecStart=/usr/local/bin/btrust
```
Comment (haslersn) [inline files/btrust.service]: TODO: Also deploy btrust using this role.

Comment (GreatTeacherOni) [inline files/btrust.service]: TODO: Deploy this binary via this role

**Review by GreatTeacherOni — DISMISSED** [review]

File: files/btrust.service [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[Unit]
```
Comment (SlothOfAnarchy) [inline files/btrust.service]:
```suggestion
# {{ ansible_managed }}
[Unit]
```

File: files/pulseaudio-bluetooth.conf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<busconfig>
```
Comment (SlothOfAnarchy) [inline files/pulseaudio-bluetooth.conf]: Please add a comment indicating this file is `ansible_managed`

File: files/pulseaudio.service [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[Unit]
```
Comment (SlothOfAnarchy) [inline files/pulseaudio.service]: Does the pulseaudio package not ship its own service file?

File: files/btrust.service [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+Description=Btrust
+After=bluetooth.service
```
Comment (SlothOfAnarchy) [inline files/btrust.service]: If this tool uses `pulseaudio`, it should be listed as dependency.

File: files/pulseaudio-bluetooth.conf [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  
+ <policy user="pulse">  
+  <allow send_destination="org.bluez"/>
```
Comment (SlothOfAnarchy) [inline files/pulseaudio-bluetooth.conf]: This should probably be configurable for when someone wants to use another Bluetooth manager.

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      - pulseaudio-module-bluetooth
+      - python-dbus
+
+- name: Create Pulse Audio service
+  copy:
+    src: pulseaudio.service
+    dest: /etc/systemd/system/pulseaudio.service
+  notify: Restart Pulse Audio
+
+- name: Give Pulse Audio permissions to use Bluetooth
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: *PulseAudio (a few times)

File: tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  copy:
+    src: pulseaudio-bluetooth.conf
+    dest: /etc/dbus-1/system.d/pulseaudio-bluetooth.conf
+  notify: Restart Pulse Audio
+
+- name: Add root to the required groups
+  user:
+    name: root
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Why `root`? As the service files are being deployed by this role, the service could easily have its own user.

File: tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    append: true
+    groups:
+      - pulse-access
+      - audio
+  notify: Restart Pulse Audio
+
+- name: Load module bluetooth-policy
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Please combine the tasks into one and loop over the modules:
```
- name: Load module bluetooth modules in PulseAudio
  lineinfile:
    dest: /etc/pulse/system.pa
    line: "load-module {{ item }}"
  notify: Restart Pulse Audio
  loop:
   - module-bluetooth-policy
   - module-bluetooth-discover
   - module-switch-on-connect
```

File: templates/btrust.toml.j2 [hunk 10]
```diff
@@ -0,0 +1 @@
+adapter_alias = "{{ a2dp_sink_adapter_name }}"
```
Comment (SlothOfAnarchy) [inline templates/btrust.toml.j2]: Please add a comment indicating this file is `ansible_managed`.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: files/btrust.service (hunk 5 above)
Comment (haslersn) [reply files/btrust.service]: Btrust is just a tool that trusts Bluetooth peers ans doesn't need pulseaudio. Pulseaudio does the actual pairing and A2DP.

File: files/pulseaudio-bluetooth.conf (hunk 6 above)
Comment (haslersn) [reply files/pulseaudio-bluetooth.conf]: pulseaudio-bluetooth is currently the only bluetooth manager which this role can install. Do you know any other a2dp bluetooth managers for linux? (bluez-alsa doesn't support it anymore.)

File: files/pulseaudio.service (hunk 4 above)
Comment (haslersn) [reply files/pulseaudio.service]: No

File: files/pulseaudio-bluetooth.conf (hunk 6 above)
Comment (SlothOfAnarchy) [reply files/pulseaudio-bluetooth.conf]: Okay I looked for it, the manager I had in mind also uses bluez.

File: tasks/main.yml (hunk 9 above)
Comment (haslersn) [reply tasks/main.yml]: Can only be done for the first two ([and is now done](https://github.com/stuvusIT/a2dp-sink/pull/1/commits/93db200515149c10be857bf48471c337f116515b)) as the third is optional.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]
Looks good except for the missing readme and the binary deployment.

**Comment by haslersn** [comment]
TODO: `README.md`

File: tasks/build_btrust.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
+---
+# This is included in deploy.yml only if {{ a2dp_sink_btrust_binary_url }} is not defined
+
+- name: Install toolchain
+  shell: curl https://sh.rustup.rs -sSf | sh -s -- -y
```
Comment (SlothOfAnarchy) [inline tasks/build_btrust.yml]: Is there no PPA or something similar for this?

Comment (haslersn) [reply tasks/build_btrust.yml]: Some distros package `rustup`, but not debian. See rust-lang/rustup.rs#800

**Pull request closed without merging**